src/main/java/DukeExceptionHandler.java

```diff
@@ -0,0 +1,99 @@
+public class DukeExceptionHandler {
+
+    /** Checks for Duke exception in the user input.
+     * @param userInput A String that represent the user's input
+     * @return A DukeException if an exception is found, null if there is no found exception
+     */
+    public static DukeException checkForException(String userInput) {
```

Comment:
This method is a little sketchy IMO, so here's some friendly suggestions!

it should make the code cleaner if you used `string.split()` or a `Scanner` to iterate through the tokens, instead of manually checking the length of the input string and doing a manual match

also, i don't think i've reached the part where you call this function yet (i'm just scrolling from the top), but it's kinda weird (imo ofc) for a method to return an exception instead of just throwing it? for this kind of paradigm i think it might be better to use `Optional` in conjunction with an error message, or just make the method throw?